packages/program/src/functions/base-invocation-scope.ts

```diff
@@ -410,6 +410,20 @@ export class BaseInvocationScope<TReturn = any> {
     return InvocationCallResult.build<T>(this.functionInvocationScopes, response, this.isMultiCall);
   }
 
+  async get<T = TReturn>(): Promise<InvocationCallResult<T>> {
+    const { receipts } = await this.getTransactionCost();
```

Comment:
Just so I understand, so we are using `getTransactionCost` to `dryRun` the transaction and getting the obtained value from the `ReturnData` receipt? Then just building the result using that?

Reply:
Exactly. Since `getTransactionCost` already executes a `dry-run` within the `estimateTxDependencies` and returns its receipts.

`InvocationCallResult` is only interested in the receipts returned by a dry-run. 

From there we will extract the TX logs, and therefore, extract the returned value from the script, or throw if a `Revert` receipt is present.

Reply:
Makes sense, I'm wonder if `getTransactionCost` should be renamed as that's doing the real work here, but happy to consider that out of scope here.

Reply:
@danielbate, well the name says (kind of) what's intended to do. 

The main purpose of `getTransactionCost` is to estimate the transaction fee. Unfortunately, this estimation involves many steps and the method does many things in one place.

I do agree that we could have a better name for it though, like: `estimateTransactionFee`